WebServer/WebServer.cpp

```diff
@@ -102,56 +102,66 @@ namespace WebServer
 		logger->Info(Languages::TextWebServerStopped);
 	}
 
-	void WebServer::LogBrowserInfo(const unsigned short port)
-	{
-		static const string Http("\n   http://");
-		static const string Port(to_string(port));
-		string localhostInfo(Http);
-		localhostInfo += "localhost:" + Port + "/";
-		string ipv4Info;
-		string ipv6Info;
-
-		struct ifaddrs* ifAddrStruct = nullptr;
-		struct ifaddrs* ifa = nullptr;
-		getifaddrs(&ifAddrStruct);
-
-		for (ifa = ifAddrStruct; ifa != nullptr; ifa = ifa->ifa_next)
-		{
-			if (!ifa->ifa_addr)
-			{
-				continue;
-			}
-			if (ifa->ifa_addr->sa_family == AF_INET)
-			{ // check it is IP4
-				// is a valid IP4 Address
-				void* tmpAddrPtr = &((struct sockaddr_in*) ifa->ifa_addr)->sin_addr;
-				char addressBuffer[INET_ADDRSTRLEN];
-				inet_ntop(AF_INET, tmpAddrPtr, addressBuffer, INET_ADDRSTRLEN);
-				string address(addressBuffer);
-				if (address.compare("0.0.0.0") == 0)
-				{
-					continue;
-				}
-				if (address.substr(0, 8).compare("169.254.") == 0)
-				{
-					continue;
-				}
-				ipv4Info += Http + addressBuffer + ":" + Port + "/";
-			}
-			else if (ifa->ifa_addr->sa_family == AF_INET6)
-			{ // check it is IP6
-				// is a valid IP6 Address
-				void* tmpAddrPtr = &((struct sockaddr_in6*) ifa->ifa_addr)->sin6_addr;
-				char addressBuffer[INET6_ADDRSTRLEN];
-				inet_ntop(AF_INET6, tmpAddrPtr, addressBuffer, INET6_ADDRSTRLEN);
-				ipv4Info += Http + "[" + addressBuffer + "]:" + Port + "/";
-			}
-		}
-		if (ifAddrStruct != NULL)
-		{
-			freeifaddrs(ifAddrStruct);
+	void WebServer::LogBrowserInfo(const std::string& listen_address, const unsigned short port)
+	{
+        static const string Http("\n   http://");
+        static const string Port(to_string(port));
+
+        if(listen_address != "::1") {
+
+            string localhostInfo(Http);
+            localhostInfo += "localhost:" + Port + "/";
+            string ipv4Info;
+            string ipv6Info;
+
+            struct ifaddrs* ifAddrStruct = nullptr;
```

Comment:
I don't like copied code. Instead of adding an if-else at line 112 it is possible to add an if without an else here until 157.